var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -248,6 +250,11 @@ def configure_args(self):
             '--with-geotiff={0}'.format(spec['libgeotiff'].prefix),
             '--with-libjson-c={0}'.format(spec['json-c'].prefix),
         ]
+        # Fix for gdal always using suffix 'lib', even though libraries
+        # like json-c might get installed in 'lib64'
+        #ldflags.append(self.spec['libtiff'].libs.search_flags)
```

Comment:
Should we do the same for `libtiff` and `libgeotiff`?

Reply:
I'm surprised that this solution is needed. Doesn't Spack automatically inject these into the compiler search path already?

Reply:
Yes it should be done automatically. I'd be interested in seeing the compiler wrapper log generated when installing this with `spack -d...`. This (error and the fix) could happen if the build bypasses the Spack compiler wrappers (in which case the wrapper never gets a chance to insert the `-L`, but setting `LDFLAGS` in configure works because the build system then arranges things properly); in that case, the compiler wrapper log will be boring (but it will be apparent that the log is missing an entry for whatever compilation line fails in the build log).

Reply:
I think https://github.com/OSGeo/gdal/issues/3895#issuecomment-850031827 has the logs you want (note that it's ~1 yr old)

Reply:
Apologies I did not make this clear but I would also need the `spack-build-out.txt` log to see which file failed, at which point I could cross-reference with the compiler wrapper log. Sometimes that is just the last entry in the log, but when I opened the log you posted there, the last entry included a `-L<json-prefix>/lib64`, so that looked good.

Also, I double checked our default library-finding logic and it appears to be capable of checking both `lib` and `lib64`.

Reply:
Scroll up to the top of that same issue for the build log (spack-build-out.txt).

Reply:
Interesting, the line which fails in the build log:

`/bin/sh /data/geol552/a/adamjs5/spack-stage-gdal-3.2.3-ozkzmsnz4b4ucfeag6ezpb7xg6e67kpf/spack-src/libtool --mode=link --silent /data/keeling/a/adamjs5/spack/lib/spack/env/gcc/g++  gdalinfo_bin.lo  /data/geol552/a/adamjs5/spack-stage-gdal-3.2.3-ozkzmsnz4b4ucfeag6ezpb7xg6e67kpf/spack-src/libgdal.la  -o gdalinfo`

Has no corresponding wrapper invocation, which suggests to me that `libtool --mode=link` is bypassing the Spack compiler wrapper (even though you refer to `/data/keeling/a/adamjs5/spack/lib/spack/env/gcc/g++`) or doing some check and deciding things won't work before it invokes it.